src/environment.ts

```diff
@@ -235,6 +236,35 @@ async function tryCreateMinGWEnvironment(dir: string): Promise<Environment | und
   return;
 }
 
+// Detect Emscripten environment
+async function tryCreateEmscriptenEnvironment(emscripten: string): Promise<Environment | undefined> {
+  let cmake_toolchain = path.join(emscripten, 'cmake', 'Modules', 'Platform', 'Emscripten.cmake');
+  if (await async.exists(cmake_toolchain)) {
+    // read version and strip "" and newlines
+    let version = fs.readFileSync(path.join(emscripten, 'emscripten-version.txt'), 'utf8');
+    version = version.replace(/["\r\n]/g, '');
+    log.verbose('Found Emscripten ' + version + ': ' + cmake_toolchain);
+    if (process.platform === 'win32') {
+      cmake_toolchain = cmake_toolchain.replace(/\\/g, path.posix.sep);
+    }
+    const ret: Environment = {
+      name: `Emscripten - ${version}`,
+      mutex: 'emscripten',
+      description: `Root at ${emscripten}`,
+      settings: {
+        'CMAKE_TOOLCHAIN_FILE': cmake_toolchain
+      },
+      variables: new Map<string, string>([]),
+      preferredGenerator: {
+        name: 'Ninja'
```

Comment:
`preferredGenerator` may be a bad name, something environment-specific generator is more accurate.
That is generator for build tool which usually shipped with particular compiler. That's why MSBuild and MinGW Makefiles are populated here. Ninja and Unix Makefiles are generic and searched first, based on `cmake.preferredGenerators` setting.

So if Emscripten doesn't ship Ninja I suggest leave this field undefined.